sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -398,8 +399,8 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
   private var numKeys = 0L
 
   // Tracking average number of probes per key lookup.
-  private var numKeyLookups = 0L
-  private var numProbes = 0L
+  private var numKeyLookups = new LongAdder
+  private var numProbes = new LongAdder
```

Comment:
I'm surprised. I think `LongToUnsafeRowMap` is used in a single thread environment and multi-thread contend should not be an issue here. Do you have any insights about how this fixes the perf issue?

Reply:
Initially, I thought these two variables in class scope will affect SIMD optimization of JIT(after java8),  we try to add `-XX: -UseSuperWord` to executor java opts to vertify this view, but no  affect with spark-2.1, although this patch can imporve performance....